llvm/lib/Transforms/Utils/ProfileVerify.cpp

```diff
@@ -63,6 +67,17 @@ bool ProfileInjector::inject() {
   // will get the same BPI it does if the injector wasn't running.
   auto &BPI = FAM.getResult<BranchProbabilityAnalysis>(F);
 
+  // Inject a function count if there's none. It's reasonable for a pass to
+  // want to clear the MD_prof of a function with zero entry count. From a
+  // metadata economy perspective, if a profile comes empty for a function, it's
+  // cheaper to assign it the 0-entry count explicitly than to mark every branch
+  // as cold.
+  if (!F.getEntryCount(true))
+    F.setEntryCount(DefaultFunctionEntryCount);
+  // If there is an entry count that's 0, then don't bother injecting. We won't
+  // verify these either.
+  if (F.getEntryCount(true)->getCount() == 0)
```

Comment:
Hoist the entry count as a separate var and reuse? Same for the usage below.

Reply:
the previous line can change the outcome here, and technically one _could_ decide to pass DefaultFunctionEntryCount == 0 (that wouldn't be the intent of the flag, and that's why I don't meant to test that, but the current code would avoid some head scratching in that case.)

Reply:
I did it though for the usage in the verifier